perception/mil_vision/include/mil_vision_lib/image_acquisition/ros_camera_stream.hpp

```diff
@@ -292,6 +292,7 @@ operator[](int i)
   {
     auto err = "ROSCameraStream: The circular buffer index you are trying to acess is out of bounds:\n"_s + e.what();
     ROS_WARN_THROTTLE_NAMED(1, "ROSCameraStream", err.c_str());
+    _mtx.unlock();
```

Comment:
This is exactly the type of thing [std::lock_guard](http://en.cppreference.com/w/cpp/thread/lock_guard) is for. We should replace the manual mutex locks/unlocks with lock guards throughout our code base.